test/e2e/api/domApi.test.js

```diff
@@ -0,0 +1,92 @@
+  });
+
+  test("Create a map viewer with document.createElement(mapml-viewer)", async () => {
+    const viewerHandle = await page.evaluateHandle(()=> document.createElement("mapml-viewer"));
+    const nn = await (await page.evaluateHandle(viewer => viewer.nodeName, viewerHandle)).jsonValue();
+    expect(nn).toEqual('MAPML-VIEWER');
+    await page.evaluateHandle((viewer) => viewer.setAttribute("lat", 45), viewerHandle);
+    await page.evaluateHandle((viewer) => viewer.setAttribute("lon", -90), viewerHandle);
+    await page.evaluateHandle((viewer) => viewer.setAttribute("zoom", 2), viewerHandle);
+    await page.evaluateHandle((viewer) => viewer.setAttribute("controls", "controls"), viewerHandle);
+    await page.evaluateHandle((viewer) => viewer.setAttribute("projection", "CBMTILE"), viewerHandle);
+    await page.evaluateHandle( (viewer) => document.body.appendChild(viewer), viewerHandle);
+    const velName = await page.evaluate(() => document.body.querySelector("mapml-viewer").nodeName);
+    expect(velName).toBe('MAPML-VIEWER');
+    // testing to ensure mapml-viewer was successfully implemented
+    let verifymap= await page.evaluate( viewer => document.body.querySelector("mapml-viewer").childElementCount);
+    expect(verifymap).toBe(1);
+    // the map, having 0 layers, should not have a layer control, despite the controls attribute
+    const layerControlHidden = await page.$eval(
+      "css=body > mapml-viewer >> css=div > div.leaflet-control-container > div.leaflet-top.leaflet-right > div",
+      (elem) => elem.hasAttribute("hidden")
+    );
+    await expect(layerControlHidden).toEqual(true);
+  });
+
+  test("Create a layer with document.createElement(layer-)", async () => {
+    const layerHandle = await page.evaluateHandle(()=> document.createElement("layer-"));
+    const nn = await  (await page.evaluateHandle(viewer => viewer.nodeName, layerHandle)).jsonValue();
+    expect(nn).toEqual('LAYER-');
+    await page.evaluateHandle((layer) => layer.setAttribute("label", "CBMT"), layerHandle);
+    await page.evaluateHandle((layer) => layer.setAttribute("src", "http://geogratis.gc.ca/mapml/en/cbmtile/cbmt/"), layerHandle);
+    await page.evaluateHandle((layer) => layer.setAttribute("checked", "checked"), layerHandle);
+    await page.evaluateHandle((layer) => document.querySelector('mapml-viewer').appendChild(layer), layerHandle);
+    let layerControlHidden = await page.$eval(
+      "css=body > mapml-viewer >> css=div > div.leaflet-control-container > div.leaflet-top.leaflet-right > div",
+      (elem) => elem.hasAttribute("hidden")
+    );
+    expect(layerControlHidden).toEqual(false);
+
+    // set the layer's hidden attribute, the layer should be removed from the layer
+    // control (but not the map), which leaves 0 layers in the layer control, which means the
+    // layer control should disappear
+    await page.evaluateHandle((layer) => layer.setAttribute("hidden", "hidden"), layerHandle);
```

Comment:
Good job on the testing, I think another test should be added similar to this, which checks for the layer control to disappear when you remove the layer element from the map.

Reply:
Same tests for web-map should also be added.